dbr quant: add test for qconfig_dict and methods

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #70200 dbr quant: start recording ops which are not quantizeable
* #70196 dbr quant: start reusing static quant module mappings
* #70128 dbr quant: better fix for bug with recursion on dequantize
* #70115 dbr quant: stop overridding tensor getters
* #70114 dbr quant: clean up logging format
* **#70109 dbr quant: add test for qconfig_dict and methods**

Summary:

Adds a test case for DBR quant + qconfig_dict specifying methods
by object_type.  Fixes a bug in the FX rewriter for scripting
to make the test pass.

Full coverage of methods will come in future PRs, this PR is
just to verify qconfig_dict is hooked up correctly.

Test plan:

```
python test/test_quantization.py TestQuantizeDBR.test_qconfig_dict_object_type_method
```

Differential Revision: [D33188160](https://our.internmc.facebook.com/intern/diff/D33188160)

vkuzo committed:
dbr quant: add test for qconfig_dict and methods

Summary:

Adds a test case for DBR quant + qconfig_dict specifying methods
by object_type.  Fixes a bug in the FX rewriter for scripting
to make the test pass.

Full coverage of methods will come in future PRs, this PR is
just to verify qconfig_dict is hooked up correctly.

Test plan:

```
python test/test_quantization.py TestQuantizeDBR.test_qconfig_dict_object_type_method
```

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33188160).

jerryzh168 commented on torch/ao/quantization/_dbr/auto_trace_rewriter.py:
```diff
@@ -100,6 +100,14 @@ def create_node(self, kind, target, args, kwargs, name=None, type_expr=None):
             target = torch.mul
 
+        # TODO(future PR): move this into mappings
+        if target == 'add':
```
guess it's OK for now. in the long term would you prefer to use torch.Tensor.add or `add`?

vkuzo replied:
I think `torch.Tensor.add` sgtm for user facing APIs.  This codepath has to match the FX representation, so this codepath probably wouldn't change.

vkuzo committed:
Update on "dbr quant: add test for qconfig_dict and methods"


Summary:

Adds a test case for DBR quant + qconfig_dict specifying methods
by object_type.  Fixes a bug in the FX rewriter for scripting
to make the test pass.

Full coverage of methods will come in future PRs, this PR is
just to verify qconfig_dict is hooked up correctly.

Test plan:

```
python test/test_quantization.py TestQuantizeDBR.test_qconfig_dict_object_type_method
```

Differential Revision: [D33188160](https://our.internmc.facebook.com/intern/diff/D33188160)

[ghstack-poisoned]
